core/typedefs.h

```diff
@@ -440,3 +440,23 @@ inline constexpr bool is_zero_constructible_v = is_zero_constructible<T>::value;
 #define GODOT_MSVC_WARNING_POP
 #define GODOT_MSVC_WARNING_PUSH_AND_IGNORE(m_warning)
 #endif
+
+template <typename T, typename = void>
+struct is_fully_defined : std::false_type {};
+
+template <typename T>
+struct is_fully_defined<T, std::void_t<decltype(sizeof(T))>> : std::true_type {};
+
+template <typename T>
+constexpr bool is_fully_defined_v = is_fully_defined<T>::value;
+
+#if !defined(SCU_BUILD_ENABLED) && !defined(DEV_ENABLED) && !defined(DEBUG_ENABLED)
```

Comment:
I see no reason to exclude debug/dev builds from these assertions; that's arguably where they'd be most useful! If a user really wants compilation to "just work," it's far more reasonable for them to either make this conditional universally `true`, or comment out a specific assertion.
```suggestion
#ifndef SCU_BUILD_ENABLED
```

Reply:
Hmm, I suppose that's true. 
Let's go with your suggestion; if it turns out to annoy contributors, we can always restrict the checking more again later.